packages/js/components/src/date-time-picker-control/test/index.tsx

```diff
@@ -659,4 +659,112 @@ describe( 'DateTimePickerControl', () => {
 
 		await waitFor( () => expect( onChangeHandler ).not.toHaveBeenCalled() );
 	} );
+
+	it( 'should not call onChange if currentDate is set to an equivalent UTC date without Zulu offset specifier', async () => {
```

Comment:
Thanks for the added tests!  Love that we're checking these more specific cases.